services/ui-src/src/components/modals/AddEditReportModal.tsx

```diff
@@ -82,9 +85,25 @@ export const AddEditReportModal = ({
     setForm(customizedModalForm);
   }, [selectedReport, copyEligibleReportsByState]);
 
+  const onChange = (event: InputChangeEvent) => {
+    if (
+      reportType === ReportType.MCPAR &&
+      event.target.name === "programName"
+    ) {
+      const customizedModalForm = structuredClone(modalFormJson);
+
+      if (event.target.value === "Other, specify") {
+        generateProgramListFields(customizedModalForm);
```

Comment:
`generateProgramListFields` is returning a value but we aren't catching that value here